zanata-war/src/main/java/org/zanata/action/VersionHomeAction.java

```diff
@@ -879,8 +887,12 @@ private void uploadAdapterFile() {
             HRawDocument rawDocument = new HRawDocument();
             rawDocument.setDocument(document);
             rawDocument.setContentHash(new String(Hex.encodeHex(md5hash)));
-            rawDocument.setType(DocumentType.typeFor(translationFileServiceImpl
-                    .extractExtension(fileName)));
+            if(docType.isPresent()) {
+                rawDocument.setType(DocumentType.getByName(docType.get()));
+            } else {
+                rawDocument.setType(DocumentType.fromSourceExtension(
+                    FileUtil.extractExtension(fileName)).iterator().next());
```

Comment:
Oh yeah, that is a tricky one - the safest is to force the user to choose one when it is ambiguous like that.